jc-1663 update connectors prereq (#602)
redhat-developer/app-services-guides: docs/connectors/getting-started-connectors/README.adoc

END GENERATED ATTRIBUTES
////

:openshift: OpenShift
:openshift-dedicated: OpenShift Dedicated

> 

Reviewer: Do not add new attributes directly to a README.adoc file. 

Instead, add the new attributes to the attributes doc (https://github.com/redhat-developer/app-services-guides/blob/main/docs/_artifacts/document-attributes.adoc), and then regenerate the attributes in all README.adoc files as described in the "Attribute injection" section of the https://github.com/redhat-developer/app-services-guides/blob/main/CONTRIBUTING.adoc file.

[id="chap-getting-started-connectors"]
= Getting started with {product-long-connectors}
ifdef::context[:parent-context: {context}]
:context: getting-started-connectors

> 

Reviewer: Probably don't need this extra blank line?

====
endif::[]

> ifndef::qs[]

Author: yes - it applies to both

[id="proc-configuring-kafka-for-connectors_{context}"]
== Verifying that you met the prerequisites for using {product-long-connectors}

[role="_abstract"]

> Before you can use {product-connectors}, you should complete the following prerequisites: 

Author: here's the rub. And it shows an issue with the product which the documentation must work around. In this case, it is not a must - people can do some of these prerequisites later but the best flow is to do them as prerequisites. That is why I use should here even though I would rather use "must" - it is not accurate

> Before you can use {product-connectors}, you should complete the following prerequisites: 

Reviewer: If some are mandatory and some are optional, that difference should be spelled out (list them in different lists)? 

If all must be done at some stage, I think it doesn't matter that you can do some of them later -- if they must be done to use the product, list them as prereqs and remove the "should"?

* Determine which OpenShift environment to use for deploying your Connectors instances.
+
For Service Preview, you have two choices:

** *Evaluation*
+

> *** Hosted on a Red Hat-owned, multi-tenant OpenShift Dedicated cluster.

Author: all good points - thanks

*** You can create four Connectors instances at a time.

> *** The evaluation environment involves 48-hour expiration windows, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64[Red Hat OpenShift Connectors Service Preview evaluation guidelines^].

Author: hmmm -  I agree "involves" is not great. "provides" implies to me that it's an advantage which is not really the case from the user point of view.  Maybe "employs, implements, utilizes, or applies" I think "applies" works

> *** The evaluation environment involves 48-hour expiration windows, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64[Red Hat OpenShift Connectors Service Preview evaluation guidelines^].

Reviewer: Alternatively, you could rephrase in a similar way to the equivalent point in the Trial section:
"Your Connectors instances expire after 48 hours."

You might want to search for occurrences of "Connectors" and replace with the attribute? (sorry for not spotting this one before now!)

** *OpenShift Dedicated trial*

*** You have access to your own OpenShift Dedicated trial environment.
*** You can create an unlimited number of Connectors instances.
*** Your OpenShift Dedicated trial cluster expires after 60 days
*** Requires a cluster administrator to install the Connectors addon, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/15a79de0-8827-4bf1-b445-8e3b3eef7b01[Adding the Red Hat OpenShift Connectors add-on to your OpenShift Dedicated trial cluster^].

ifdef::qs[]
* Complete the steps in link:https://console.redhat.com/application-services/learning-resources?quickstart=getting-started[Getting started with {product-long-kafka}] quick start to set up the following components:

** A *Kafka instance* that you can use for {product-connectors}.
** A *Kafka topic* to store messages sent by data sources and make the messages available to data sinks.
** A *service account* that allows you to connect and authenticate your {connectors} instances with your Kafka instance.

> ** *Access rules* for the service account that defines how your {connectors} instances can access and use the topics in your Kafka instance.

Author: yes!

endif::[]

ifndef::qs[]
* Complete the steps in _{base-url}{getting-started-url-kafka}[Getting started with {product-long-kafka}^]_ to set up the following components:

** A *Kafka instance* that you can use for {product-connectors}.
** A *Kafka topic* to store messages sent by data sources and make the messages available to data sinks.
** A *service account* that allows you to connect and authenticate your {connectors} instances with your Kafka instance.

> ** *Access rules* for the service account that defines how your {connectors} instances can access and use the topics in your Kafka instance.

Reviewer: Would it be better to conditionalise only the lead-in sentence that's different, so that you don't have to repeat the 4 sub-bullet points that are identical? I'm not very familiar with AsciiDoc, so maybe it's not possible to do it in that way, or maybe it's preferred to conditionalise at the larger level.

Author: yes - good suggestion

endif::[]

ifdef::qs[]
.Procedure
Make sure that you have set up the prerequisite components.

.Verification

> * If you want to use a 60-day OpenShift Dedicated trial cluster to deploy your Connectors instances, did a cluster administrator add the Connectors addon to your trial cluster?

Reviewer: ```suggestion
* If you plan to use a 60-day OpenShift Dedicated trial cluster to deploy your Connectors instances, has a cluster administrator added the Connectors add-on to your trial cluster?
```

> * Is the Kafka instance listed in the Kafka instances table and is it in the *Ready* state?

Reviewer: ```suggestion
* Is the Kafka instance listed in the Kafka instances table, and is the Kafka instance in the *Ready* state?
```

> * Did you verify that your service account was successfully created in the *Service Accounts* page?

Reviewer: ```suggestion
* Is your service account successfully created in the *Service Accounts* page?
```

Before you use {product-connectors}, you should complete the following prerequisites: 

* Determine which {openshift} environment to use for deploying your Connectors instances.

* Configure {product-long-kafka} for use with Connectors.

> *Determining which {openshift} environment to use for deploying your Connectors instances*

Author: I tried that - looked odd

For Service Preview, you have two choices:

* *The hosted evaluation environment*

** The Connectors instances are hosted on a multitenant {openshift-dedicated} cluster that is owned by Red Hat.
** You can create four Connectors instances at a time.
** The evaluation environment applies 48-hour expiration windows, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64[Red Hat {openshift} Connectors Service Preview evaluation guidelines^].

* *Your own trial environment*

** You have access to your own {openshift-dedicated} trial environment.
** You can create an unlimited number of Connectors instances.
** Your {openshift-dedicated} trial cluster expires after 60 days.
** A cluster administrator must install the Connectors add-on as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/15a79de0-8827-4bf1-b445-8e3b3eef7b01[Adding the Red Hat OpenShift Connectors add-on to your {openshift-dedicated} trial cluster^].

> *Configuring {product-long-kafka} for use with Connectors*

Author: I tried that - looked odd

ifndef::qs[]
.Verification

> * If you want to use a 60-day OpenShift Dedicated trial cluster to deploy your Connectors instances, verify that a cluster administrator added the Connectors addon to your trial cluster.

Reviewer: ```suggestion
* If you plan to use a 60-day OpenShift Dedicated trial cluster to deploy your Connectors instances, verify that a cluster administrator added the Connectors add-on to your trial cluster.
```

endif::[]

.Procedure

> . In the {product-long-connectors} web console, select *{connectors} * and then click *Create {connectors} instance*.

Reviewer: ```suggestion
. In the {product-long-connectors} web console, select *{connectors}* and then click *Create {connectors} instance*.
```

. Review the summary of the configuration properties and then click *Create {connectors} instance*.
+

> Your {connectors} instance is listed in the table of {connectors} .

Reviewer: ```suggestion
Your {connectors} instance is listed in the table of {connectors}.
```
